drivers/at86rf2xx/at86rf2xx_getset.c

```diff
@@ -150,23 +151,17 @@ void at86rf2xx_set_addr_short(at86rf2xx_t *dev, uint16_t addr)
                         dev->netdev.short_addr[0]);
 }
 
-uint64_t at86rf2xx_get_addr_long(const at86rf2xx_t *dev)
+void at86rf2xx_get_addr_long(const at86rf2xx_t *dev, void *addr)
 {
-    uint64_t addr;
-    uint8_t *ap = (uint8_t *)(&addr);
-
-    for (int i = 0; i < 8; i++) {
-        ap[i] = dev->netdev.long_addr[i];
-    }
-    return addr;
+    memcpy(addr, dev->netdev.long_addr, IEEE802154_LONG_ADDRESS_LEN);
 }
 
-void at86rf2xx_set_addr_long(at86rf2xx_t *dev, uint64_t addr)
+void at86rf2xx_set_addr_long(at86rf2xx_t *dev, const void *addr)
 {
-    for (int i = 0; i < 8; i++) {
-        dev->netdev.long_addr[i] = (uint8_t)(addr >> (i * 8));
+    memcpy(dev->netdev.long_addr, addr, IEEE802154_LONG_ADDRESS_LEN);
+    for (unsigned i = 0; i < IEEE802154_LONG_ADDRESS_LEN; i++) {
         at86rf2xx_reg_write(dev, (AT86RF2XX_REG__IEEE_ADDR_0 + i),
-                            (addr >> ((7 - i) * 8)));
+                            dev->netdev.long_addr[i]);
```

Comment:
Isn't this now in wrong order? The device expects the address to be LSB first, right?

Reply:
I'm honestly unsure about this, the bit-shifting that is removed here makes things very much unclear ^^"

Reply:
Yes indeed. And I have been wrong about the IEEE address being a byte array. Maybe I should stick with the application layer. But I truly fail to understand, why is it was defined as a 64 bit number and not as 8 octets and therefore avoiding the confusion altogether...

Reply:
I'll test tomorrow to be sure pinging still works. It's too late for me to think straight.

Reply:
> But I truly fail to understand, why is it was defined as a 64 bit number and not as 8 octets and therefore avoiding the confusion altogether...

Me neither, but apparently the original contributor thought it to be simple this way.